yt_dlp/extractor/yandexvideo.py

```diff
@@ -234,7 +235,7 @@ def _real_extract(self, url):
         id = self._match_id(url)
         webpage = self._download_webpage(url, id)
         data_json = self._parse_json(
-            self._search_regex(r'data\s*=\s*({["\']_*serverState_*video.+?});', webpage, 'metadata'), id)
+            self._search_regex(r'(?m)data\s*=\s*({["\']_*serverState_*video.+?});$', webpage, 'metadata'), id)
```

Comment:
why?

Reply:
I ran into a case where the first form did not work while the second did. I think it was because the text line was something insane like over 2 million long and the parser was giving up early. Using a multi-line regex worked. Or the JS code was simply written over multiple text lines. I'm not super sure I remember the exact reason.

Reply:
It seems to work fine without it. If you can find a counter example, add a test with it. I'm not adding this if even you don't know what it does!